app/src/main/java/org/systers/mentorship/remote/TokenAuthenticator.kt

```diff
@@ -18,9 +18,15 @@ class TokenAuthenticator: Authenticator{
     override fun authenticate(route: Route, response: Response): Request? {
         if (response.code() == 401) {
             preferenceManager.clear()
+
+            if(response.request().url().encodedPath() == "/login" || response.request().url().encodedPath() == "/register")
+                return null
+
             val intent = Intent(MentorshipApplication.getContext(), LoginActivity::class.java)
             intent.putExtra(Constants.TOKEN_EXPIRED_EXTRA, 0)
             ContextCompat.startActivity(MentorshipApplication.getContext(), intent, null)
+
+            return null
```

Comment:
Also why are you returning null here?

Reply:
returning null here is necessary, otherwise the bug for which this issue has been created will still persist in case when auth token expires in the app other than in login and signup activities